packages/types/index.d.ts

```diff
@@ -145,9 +145,9 @@ export interface IFlow {
 export interface IUser {
   id: string
   email: string
-  connections: IConnection[]
-  flows: IFlow[]
-  steps: IStep[]
+  connections?: IConnection[]
+  flows?: IFlow[]
+  steps?: IStep[]
```

Comment:
oh why does this become optional suddenly ah? is it following the `user.ts` model

Reply:
Yep, our `user` model doesn't guarantee that connections, flows and steps are fetched, so we can't make it required props.

Reply:
Also as a side comment, I feel like there's something broken about this.

In theory having a common type library between frontend and backend seems to make sense, but it feels like this file has only added more confusion (via layers of indirection) and inaccurate typing.

My gut feel is that there's a bunch of casting going on to force GraphQL and nodejs types to interop nicely. Instead of using this file, we should employ GraphQL codegen to properly share types (something I wanna get everyone to look at in Q1)

Reply:
okayy, with codegen, when we do graphql queries on the frontend, it wouldn't be typed to `any` alr right

Reply:
yep yep